### ISSUE TITLE
Highlighting with mouse causes super fast scrolling

### ISSUE THREAD

**Issue by Karuption** [Jozott00/AdaptiveCaretScroll#1]
If you are highlighting a section of code while using this plugin and you highlight (with the mouse) someplace that the adaptive scroll would scroll, it is very difficult to highlight what you are trying to. This is most noticeable if you are centering the caret.  Everything works as expected when doing keyboard bound controls. 

I would expect that scrolling is significantly slower than it is while the mouse is dragging. Doing a delay for this would be optimal, but a quick glance at the JB scrolling API shows that they don't really have a delay that can be passed in. What do you think about disabling (or giving the config option to disable) the scrolling logic while the mouse is being dragged? This seems like it would solve the issue. This is what the vim emulator seemingly does and does not scroll on release.

**Comment by Jozott00** [comment]
Thanks for pointing out the issue! I agree, the scrolling behavior when highlighting with the mouse could be better, especially when the caret is centered.

I've tried to find a way to check if the mouse is pressed directly within the caret position change, but no luck so far. I did try using the MouseAdapter API to listen for mouse presses and releases. The catch is that the caret position change happens before the mouse press event is detected, so it doesn't prevent the scroll when you click somewhere.

What do you think? Should the scroll be prevented when the user clicks, or is the current behavior okay fine? 

PS: The version shown in the video is on the `feature/prevent_on_drag` branch.

https://github.com/Jozott00/AdaptiveCaretScroll/assets/12057307/b9c8a24c-aae1-4e2f-94b6-382a3b4ab6e5

**Comment by Karuption** [comment]
@Jozott00 I appreciate the responsiveness. I checked out what you did and tweaked it a little on a clone of the repo. I used the JB EditorMouseListener instead of the java one. I registered it right along what you already had and came up with this implementation:

``` kotlin    
    override fun mousePressed(event: EditorMouseEvent) {
        this.isMousePressed = true
        super.mousePressed(event)
    }

    override fun mouseReleased(event: EditorMouseEvent) {
        super.mouseReleased(event)
        this.isMousePressed = false

        if(event.editor.selectionModel.hasSelection() || event.editor.caretModel.caretCount != 1)
            return

        var shift = if (event.editor.caretModel.logicalPosition.line < event.editor.document.lineCount-1) 1 else -1
        event.editor.caretModel.currentCaret.moveCaretRelatively(0, shift, false, false)
        event.editor.caretModel.currentCaret.moveCaretRelatively(0, -shift, false, false)
    }
 ```

Since the editor mouse listener gives you access to the editor the event was in, I guard for a selection or multiple carets. I image that you may click to add a caret. I tried to set the caret to the current location, but it wouldn't fire the caret event. I was able to force the event by moving the caret with the caret model. I can PR what I have, if you want.

I really don't like doing it this way, but its the only way I could force the event. It may be worthwhile to not reset the isMousePressed until you are about to return from the if and right before the last caret move. This would minimize some scrolling and maybe make it slightly smoother.

**Comment by Jozott00** [comment]
@Karuption you are right, I used the wrong listener (oops). I would appreciate a PR!

The workaround makes kinda sense, have to try it out on the PR.

**Comment by Jozott00** [comment]
Implemented in commit ebe3453bfad7f33e327e88b87a7440647fd57b60

**Comment by Jozott00** [comment]
Just released the new version (1.0.1) that fixes the issue.
